plugins/ui-update/components/UIDetails.tsx

```diff
@@ -0,0 +1,217 @@
+    map(result => result.data.ui),
+    catchError(() => of(DEFAULT_UI_METADATA))
+  );
+}
+
+function onClickRollbackUI(
+  _e: React.MouseEvent<HTMLElement> | null,
+  delayMs: number = 45000
+) {
+  uiServiceActions$.next({
+    type: UIActionType.Reset,
+    action: UIActions.Started,
+    value: ""
+  });
+  graphqlObservable(
+    gql`
+      mutation {
+        resetDCOSUI
+      }
+    `,
+    uiServiceSchema,
+    {}
+  )
+    .pipe(
+      take(1),
+      delay(delayMs)
+    )
+    .subscribe(
+      (result: UIRollbackResult) => {
+        uiServiceActions$.next({
+          type: UIActionType.Reset,
+          action: UIActions.Completed,
+          value: result.data.resetDCOSUI || ""
+        });
+      },
+      (error: MutationError) => {
+        uiServiceActions$.next({
+          type: UIActionType.Reset,
+          action: UIActions.Error,
+          value: error.message
+        });
+      }
+    );
+}
+
+function onClickStartUpdate(version: string, delayMs: number = 45000) {
+  uiServiceActions$.next({
+    type: UIActionType.Update,
+    action: UIActions.Started,
+    value: version
+  });
+  graphqlObservable(
+    gql`
+      mutation {
+        updateDCOSUI(newVersion: $version)
+      }
+    `,
+    uiServiceSchema,
+    {
+      version
+    }
+  )
+    .pipe(
+      take(1),
+      delay(delayMs)
+    )
+    .subscribe(
+      (result: UIUpdateResult) => {
+        uiServiceActions$.next({
+          type: UIActionType.Update,
+          action: UIActions.Completed,
+          value: result.data.updateDCOSUI
+        });
+      },
+      (error: MutationError) => {
+        uiServiceActions$.next({
+          type: UIActionType.Update,
+          action: UIActions.Error,
+          value: error.message
+        });
+      }
+    );
+}
+
+const UIDetails = componentFromStream(() => {
+  const cosmosVersions$ = queryCosmosForUIVersions();
+  const uiMetadata$ = queryUIServiceForMetadata();
+  const uiMetadataOnAction$ = uiServiceActions$.pipe(
+    filter(
+      uiAction =>
+        uiAction.action === UIActions.Completed ||
+        uiAction.action === UIActions.Error
+    ),
+    switchMap(() => queryUIServiceForMetadata())
+  );
+
+  return combineLatest<[Package, UIMetadata, UIAction]>([
+    cosmosVersions$,
+    merge(uiMetadata$, uiMetadataOnAction$),
+    uiServiceActions$
+  ]).pipe(
+    map(([packageInfo, uiMetadata, uiAction]) => {
+      if (!packageInfo || !uiMetadata) {
+        return <InitialScreen />;
+      }
+      const latestUpdateAvailable = versionUpdateAvailable(
+        packageInfo,
+        uiMetadata
+      );
+
+      return (
+        <ConfigurationMapSection>
+          <InstalledVersion
+            uiMetaData={uiMetadata}
+            uiAction={uiAction}
+            onRollbackClick={onClickRollbackUI}
+          />
+          <AvailableUpdateRow
+            latestVersion={latestUpdateAvailable}
+            uiAction={uiAction}
+            onUpdateClick={onClickStartUpdate}
```

Comment:
What do  you think about adopting the convention to name callback functions "onXXX" but the handlers that are attached to them as "handleXXX"?